Admin/AbstractAdmin.php

```diff
@@ -1202,9 +1203,20 @@ public function attachAdminClass(FieldDescriptionInterface $fieldDescription)
         $adminCode = $fieldDescription->getOption('admin_code');
 
         if ($adminCode !== null) {
-            $admin = $pool->getAdminByAdminCode($adminCode);
+            try {
+                $admin = $pool->getAdminByAdminCode($adminCode);
+            } catch (ServiceNotFoundException $e) {
+                throw new \RuntimeException('Unable to find the Admin instance', $e->getCode(), $e);
+            }
         } else {
-            $admin = $pool->getAdminByClass($fieldDescription->getTargetEntity());
+            $targetEntity = $fieldDescription->getTargetEntity();
+            $admin = $pool->getAdminByClass($targetEntity);
+            if ($targetEntity && !$admin) {
+                throw new \RuntimeException(sprintf(
+                    'Unable to find a valid admin for the class: `%s`',
+                    $targetEntity
+                ));
+            }
         }
 
         if (!$admin) {
```

Comment:
As `attachAdminClass ` is a public method, we don't know all the places where it can be called. 

But for the scenario when there is no `admin_code` and no `targetEntity`, removing this line would allow to execute `setAssociationAdmin` on line [https://github.com/sonata-project/SonataAdminBundle/blob/3.x/Admin/AbstractAdmin.php#L1324](1218) which expects a `AdminInterface` and does not allow for null values. This is why I left the check as is.